os/arch/arm/src/amebasmart/amebasmart_spi.c

```diff
@@ -105,6 +105,7 @@ struct amebasmart_spidev_s {
 	uint8_t mode;               /* Mode 0,1,2,3 */
 	int role;
 	gpio_t gpio_cs0;
+	gpio_t gpio_rs;
```

Comment:
Could you please let us know why rs is added to spidev structure. Generally spi only has mosi, miso, sclk and cs. 

![Screenshot (2)](https://github.com/Samsung/TizenRT/assets/78739211/a6caca6a-c9a3-4b48-97db-daa1d9c801ca)

And more over, rs seems to be lcd specific. So, could you please check if its better to move the logic related to rs out of amebasmart_spi.c?

Reply:
This pin is required to identify the data type needed by LCD. We are using below function every time to switch it from 0 to 1 based on data type(Command/data) before sending actual data from driver.
int amebasmart_spi1cmddata(FAR struct spi_dev_s *dev, uint32_t devid, bool cmd) {
	FAR struct amebasmart_spidev_s *priv = (FAR struct amebasmart_spidev_s *)dev;
	if (cmd == 0) {
		gpio_write(&priv->gpio_rs, 0);
	} else {
		gpio_write(&priv->gpio_rs, 1);
	}
	return 0;
}
Where should this function be present if not here?

Reply:
so you mean it is send command to spi, right? then you can use cs0.
```
#st7789.h
struct st7789_config_s {
	gpio_t gpio_cs;
	gpio_t gpio_reset;
	int (*reset)(gpio_t gpio);
};

#rtl8730csm_st7789.c

struct st7789_config_s g_rtl8730_config_dev_s {
	.gpio_cs.pin = PA_10,
	.gpio_reset.pin = PB_11,
	.reset = rtl8730_st7789_gpio_reset
};

static int rtl8730_st7789_gpio_reset(gpio_t gpio)
{
	gpio_write(gpio, GPIO_SET);
	DelayMs(10);
	gpio_write(gpio, GPIO_UNSET);
	DelayMs(10);
	gpio_write(gpio, GPIO_SET);
}

void rtl8730_st7789_initialize(void)
{
    /* Please check pinmux setting necessary or not???? */
	FAR struct spi_dev_s *spi = up_spiinitialize(LCD_SPI_PORT);
	if (!spi) {
		lldbg("spi init failed\n");
		return;
	}
	struct lcd_dev_s *dev = st7789_lcdinitialize(spi, g_rtl8730_config_dev_s);
	if (lcddev_register(dev) < 0) {
		lldbg("LCD driver register fail\n");
	} else {
		lldbg("LCD driver register success\n");
	}
}

#st7789.c
struct st7789_dev_s {
	struct lcd_dev_s dev;
	st7789_config_s *config;
	/* Private LCD-specific information follows */

	FAR struct spi_dev_s *spi;  /* SPI device */
	uint8_t bpp;                /* Selected color depth */
	uint8_t power;              /* Current power setting */
	uint16_t runbuffer[ST7789_LUT_SIZE];
};


FAR struct lcd_dev_s *st7789_lcdinitialize(FAR struct spi_dev_s *spi, st7789_config_s *config)
{
	FAR struct st7789_dev_s *priv = &g_lcddev;
	/* Initialize the driver data structure */
	priv->dev.getvideoinfo = st7789_getvideoinfo;
	priv->dev.getplaneinfo = st7789_getplaneinfo;
	priv->dev.getpower     = st7789_getpower;
	priv->dev.setpower     = st7789_setpower;
	priv->dev.getcontrast  = st7789_getcontrast;
	priv->dev.setcontrast  = st7789_setcontrast;
	priv->dev.init         = st7789_init;
	priv->dev.clear        = st7789_clear;
	priv->spi              = spi;
	priv->config = config;
	priv->spi.gpio_cs0	= config->gpio_cs;
	SPI_SETMODE(spi, CONFIG_LCD_ST7789_SPIMODE);
	SPI_SETBITS(spi, LCD_ST7789_SPI_BITS);
	SPI_SETFREQUENCY(spi, CONFIG_LCD_ST7789_FREQUENCY);

	st7789_init_cmd(priv);	
	st7789_sendcmd(priv, ST7789_DISPOFF);
	st7789_bpp(priv, ST7789_BPP);
	st7789_sendcmd(priv, ST7789_DISPON);
	st7789_fill_frame(priv, 0xFFFF, 0, 0, ST7789_XRES-1, ST7789_YRES-1);
	/* For reset */
	priv->config->reset(priv->config->gpio_reset);
	
	return &priv->dev;
}
```
@abhishek-samsung please correct this if anything is wrong

Reply:
Can we use gpio_t in driver side code directly? I couldn't find any example for this.
#st7789.h
struct st7789_config_s {
	gpio_t gpio_cs;
	gpio_t gpio_reset;
	int (*reset)(gpio_t gpio);
};